streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -490,6 +494,19 @@ abstract class DStream[T: ClassTag] (
     logDebug("Cleared checkpoint data")
   }
 
+  private[streaming] def readyToShutdown(): Unit = {
+    _readyToShutdown = true
+    dependencies.foreach(_.readyToShutdown())
+    logDebug("Ready to shutdown")
```

Comment:
It's in debug level, and you can refer to other method in this file, they all have the same behavior. I think it should be fine.